RedisEnterprise-DNS-Migration/migration_script.ps1

```diff
@@ -0,0 +1,325 @@
+# Function definitions
+function Login($SubscriptionId) {
+    $context = Get-AzContext
+
+    if (!$context -or ($context.Subscription.Id -ne $SubscriptionId)) {
+        Write-Host "Connecting to Azure..."
+        Connect-AzAccount -Subscription $SubscriptionId
+    } 
+    else {
+        Write-Host "Context set to: SubscriptionId '$SubscriptionId'"
+    }
+}
+
+# This function is meant to avoid making assumptions about what resource group the user wants to create their private dns zone in. 
+# It suggests a default and error checks for user so they can pick a correct value
+function PromptForInputs-New-AzPrivateDnsZone {
+    Param(
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Name,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $ResourceGroupName,
+        [bool] $IncludeDefault = $true
+    )
+    $rgNames = (Get-AzResourceGroup).ResourceGroupName 
+    $lowerRgNames = $rgNames | ForEach-Object { $_.toLower() }
+
+    $default = $ResourceGroupName
+    if($IncludeDefault) {
+        Write-Host "Enter the resource group name for '$Name' (default: [$default])" -ForegroundColor Blue
+    } else {
+        Write-Host "Enter the resource group name for '$Name'" -ForegroundColor Blue
+    }
+    if (!($value = Read-Host )) { 
+        if($IncludeDefault) {
+            $value = $default 
+        } else {
+            $value = $null
+        }
+    }
+    if ($null -eq $value -or !($lowerRgNames -contains $value.ToLower())) {
+        Write-Host "'$value' is not an existing resource group. Resource group must be one of the following: $($rgNames -join ", ")" -ForegroundColor Blue
+        return PromptForInputs-New-AzPrivateDnsZone $Name $ResourceGroupName $IncludeDefault
+    }
+    # Determine if one already in this resource group
+    $existingZone = $null
+    try {
+        $existingZone = Get-AzPrivateDnsZone -ResourceGroupName $value -Name $Name -ErrorAction SilentlyContinue
+    } catch {}
+    if (!($null -eq $existingZone)) {
+        $existingLinks = Get-AzPrivateDnsVirtualNetworkLink -ResourceGroupName $existingZone.ResourceGroupName -ZoneName $existingZone.Name 
+        if($null -eq $existingLinks) {
+            Write-Host "An existing zone with the name '$Name' was already found in the resource group '$value', but it has no existing network links. Using that one instead." -ForegroundColor Yellow
+            return $existingZone
+        } else {
+            Write-Host "An existing zone with the name '$Name' was already found in the reosurce group '$value' and is linked to vnet $($existingLinks.VirtualNetworkId)." -ForegroundColor Yellow
+            Write-Host "This zone can be linked to multiple vnets if there is no cache that needs to be accessed from more than one of the vnets. This limitation is because otherwise the DNS A records will clash." -ForegroundColor Yellow
```

Comment:
would love to see VNets throughout, but it is strictly 'nice to have' . I won't comment each one.
```diff
- This zone can be linked to multiple vnets
+ This zone can be linked to multiple VNets
```

Reply:
```diff
-This limitation is because
+This limitation exists because otherwise,

```